dash_auth/basic_auth.py

```diff
@@ -15,11 +15,9 @@ def is_authorized(self):
         username_password = base64.b64decode(header.split('Basic ')[1])
         username_password_utf8 = username_password.decode('utf-8')
         username, password = username_password_utf8.split(':')
-        for pair in self._username_password_list:
-            if pair[0] == username and pair[1] == password:
-                return True
-
-        return False
+        return (username, password) in (
+            (x[0], x[1]) for x in self._username_password_list
+        )
```

Comment:
Sorry but this does not really improve it. It should use a dict instead:

```python
    def __init__(self, app, username_password_list):
        Auth.__init__(self, app)
        self._users = username_password_list \
            if isinstance(username_password_list, dict) \
            else {k: v for k, v in username_password_list}
        
    def is_authorized(self):
        ...
        return self._users.get(username) == password
```

Reply:
Hmm, I've never seen your version of BasicAuth.__init__. Is that from a dev branch or something?

Reply:
I just wrote it...

Reply:
Sounds good. Dictionaries would indeed be faster for checking the password. We should decide and then make everything, as `username_password_list` is shown as a list of lists on the Dash website, as a tuple of tuples on the GitHub example, and then as a dictionary here.